src/Microsoft.Extensions.Http/HttpClientFactoryOptions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Microsoft.Extensions.Http
+{
+    public class HttpClientFactoryOptions
+    {
+        /// <summary>
+        /// Gets a list of operations used to configure an <see cref="HttpMessageHandlerBuilder"/>.
+        /// </summary>
+        public IList<Action<HttpMessageHandlerBuilder>> HandlerBuilderActions { get; } = new List<Action<HttpMessageHandlerBuilder>>();
+
+        /// <summary>
+        /// Gets a list of operations used to configure an <see cref="HttpClient"/>.
+        /// </summary>
+        public IList<Action<HttpClient>> HttpClientActions { get; } = new List<Action<HttpClient>>();
```

Comment:
This type is actually kinda gross - but I think this is the best approach. See the commit comments